### PR TITLE
Allow initial state function for AoT compatibility

### LINKED ISSUE #51: Accept initialState function to allow dynamic initial state in AoT
This is problem carried over from @ngrx/store (see: [273](https://github.com/ngrx/store/issues/273), [348](https://github.com/ngrx/store/issues/348)).

AoT requires static objects in module definitions, so dynamic initial state is only possible with:

1. At least support for an initialState function
or
2. Putting a meta-reducer on top of everything for this purpose alone (as described in my comment on 348). The second case seems unnecessarily unnecessarily complicated for something so easy to support in the library itself.

### PR DESCRIPTION
**Pull request by bfricka**
resolves #51

### CODE REVIEW

File: docs/store/api.md [hunk 1]
```diff
@@ -20,6 +20,27 @@ import { reducers } from './reducers';
...
 
+### Initial State and Ahead of Time (AoT) Compilation
+
+Angular AoT requires all symbols referenced in the construction of its types (think `@NgModule`, `@Component`, `@Injectable`, etc.) to be statically defined. For this reason, we cannot dynamically inject state at runtime with AoT unless we provide `initialState` as a function. Thus the above `NgModule` definition simply changes to:
+
+```ts
+/// Pretend this is dynamically injected at runtime
+const initialStateFromSomewhere = { counter: 3 };
+
+/// Static static
```
Comment (littleStudent) [inline docs/store/api.md]: should be `Static state` i guess.

Comment (bfricka) [inline docs/store/api.md]: Fixed thanks

**Comment by bfricka** [comment]
FYI, tests are all passing, but coveralls is throwing a fit.

File: docs/store/api.md [hunk 2]
```diff
@@ -20,6 +20,27 @@ import { reducers } from './reducers';
...
+
+```ts
+/// Pretend this is dynamically injected at runtime
+const initialStateFromSomewhere = { counter: 3 };
+
+/// Static state
+const initialState = { counter: 2 };
+
+/// In this function dynamic state slices, if they exist, will overwrite static state at runtime.
+const getInitialState = () => ({...initialState, ...initialStateFromSomewhere});
```
Comment (brandonroberts) [inline docs/store/api.md]: This should be exported and a plain function instead of a arrow function for AOT support

Comment (bfricka) [inline docs/store/api.md]: Fixed

**Comment by brandonroberts** [comment]
Thanks!

**Pull request merged**